test/spec/generate_spec.py

```diff
@@ -110,7 +112,24 @@ def parse_directory_tree(directory):
 parser.add_argument("-r", "--root-dir", help="Path that all paths should be relative to")
 args=parser.parse_args()
 
+output_capture = io.StringIO()
+sys.stderr = output_capture
+
 directories=parse_directory_tree(args.directory)
+error_string=""
 for directory in directories:
+    output_capture.truncate(0)
+    output_capture.seek(0)
     test_case=TestCase(directory, args.root_dir)
     test_case.generate_specification()
+    if len(output_capture.getvalue()) > 0:
+        error_string=output_capture.getvalue()
+        break
```

Comment:
Minor, variable assignments in PEP-8 should be `a = b`, whereas parameter assignments in methods `a=b`.